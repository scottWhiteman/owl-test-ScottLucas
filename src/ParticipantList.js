import React from 'react';
import Participant from './Participant';
import store from './store';

class ParticipantList extends React.Component {
  render() {
    const participants = store.participants.map(participant => {
      return <Participant 
        id={participant.id}
        avatar={participant.avatar}
        name={participant.name}
        inSession={participant.inSession}
        onStage={participant.onStage}
      />
    });
    
    return (
      <ul className="ParticipantList">
          {participants}
      </ul>
    );
  }
}

export default ParticipantList;