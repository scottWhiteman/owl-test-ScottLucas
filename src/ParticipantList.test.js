import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './Participant';

it ("Participant List render without crash", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <ParticipantList />
    , div);
    ReactDOM.unmountComponentAtNode(div);
});

it ("Participants render without crash", () => {
    const tree = renderer.create(<ParticipantList />).toJSON();
    expect(tree).toMatchSnapshot();
});
