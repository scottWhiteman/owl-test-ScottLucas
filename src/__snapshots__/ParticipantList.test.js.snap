// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants render without crash 1`] = `
<li
  className="Participant"
>
  <img
    alt="this user's avatar"
  />
  <div
    className="UserStatus"
  >
    <p />
    <p>
      left session
    </p>
  </div>
</li>
`;
